client/src/main/java/io/pravega/client/connection/impl/CommandEncoder.java

```diff
@@ -56,7 +54,9 @@
     static final int MAX_QUEUED_EVENTS = 500;
     @VisibleForTesting
     static final int MAX_QUEUED_SIZE = 1024 * 1024; // 1MB
-    
+    @VisibleForTesting
+    static final int MAX_SETUP_SEGMENTS_SIZE = 2000;
```

Comment:
Ho do we know that this is the right limit? Could this lead to OOM anyway for clients with low memory resources, or could it limit the parallelism for powerful clients? I assume that this only impacts to real TCP connections and not to Connection Pooling, right?

Reply:
2k is just a number that I assume that should be fine. 10k should also work. I think making it 2000 limits the memory usage for a single TCP client to be less than 1 MB. The OOM is found when the acutal setup_segment_size being a few millions.
As for the parallelism, it means there are 2k active flows on a single TCP connection, the connection pool has logic to retrieve the connection with least flow, so that means basically every connection in the pool has 2k flows, the current default max connection for segmetstore is 10, so that's 20k active flows to a single segmentstore. I'm not sure if that's enough to use, but we can also increase the max connection for segmentstore number if we are expecting such huge workload. 
Yes it closes the TCP connection when that 2k limit is reached, so connection pool will detect that and prune this connection and perhaps creates a new one.
What do you think? Thanks!